apps/desktop-wallet/src/features/tokenPrices/tokenPricesHooks.tsx

```diff
@@ -59,6 +69,41 @@ export const useAlphPrice = () => {
   return tokenPrices.find((token) => token.symbol === ALPH.symbol)?.price
 }
 
+export const useAlphHistoricalPrices = () => {
+  const currency = useAppSelector((s) => s.settings.fiatCurrency).toLowerCase()
+
+  const { data: alphPriceHistory } = useQuery({
+    queryKey: ['tokensPrice', 'historicalPrice', ALPH.symbol, currency],
```

Comment:
I notice that you use the `tokensPrice` key here, and you used `tokenPrices` in the query above. Is this intentional? 

Using the same base key could eventually be useful to leverage `useIsFetching` ([docs](https://tanstack.com/query/latest/docs/framework/react/reference/useIsFetching))

Reply:
It was not intentional. Maybe this is a good reason to have a way to not make this mistake. Maybe something like key generator? :/

Also, pretty cool feature the one you shared

Reply:
pushed an update, lemme know what you think

Reply:
very good catch btw

Reply:
That's the kind of mistake I was talking about when talking about query collections. The more you duplicate / extend a query, the more this can happen. 
We could take another look at TkDodo's recommended approach to "query factories" https://tkdodo.eu/blog/the-query-options-api#query-factories

I personally find it a bit confusing to mix key-only values with functions, so I decided to use function only. But yeah, let's find the best solution :)

Reply:
About the "key generator" idea, I'm aligned with TkDodo:

<img width="1001" alt="image" src="https://github.com/user-attachments/assets/bb2b40d7-9644-4b86-b135-d862cbff9e02">

Reply:
I noticed that in your query collections you don't reuse query keys. The idea that queries of the same collection should share the same keys, going from the most general to the most specific (iiuc).

I agree that we should not separate the query key from the query function and I also don't like TkDodo's (old) approach on key factories: https://tkdodo.eu/blog/effective-react-query-keys#use-query-key-factories

I extracted the main query key. WDYT?

Reply:
My guess is that combining a simple map of query keys (similar to what you did) close to the query collection (what I did) could be a good solution.
But let's see further down the road!